pkg/router/template/router.go

```diff
@@ -506,6 +527,8 @@ func (r *templateRouter) AddEndpoints(id string, endpoints []Endpoint) bool {
 		return false
 	}
 
+	r.lock.Lock()
```

Comment:
Hmm, might be better to separate FindServiceUnit into an internal/external part - that way all the internal calls can work thru' in a lock. Let me redo this. Thx